packages/mask/src/extension/popups/pages/Personas/hooks/usePersonaContext.ts

```diff
@@ -2,32 +2,37 @@ import { createContainer } from 'unstated-next'
 import { useValueRef } from '@masknet/shared-base-ui'
 import { ECKeyIdentifier, EMPTY_LIST, PersonaInformation } from '@masknet/shared-base'
 import { currentPersonaIdentifier } from '../../../../../settings/settings'
-import { useAsync, useAsyncRetry } from 'react-use'
+import { useAsyncRetry } from 'react-use'
 import Services from '../../../../service'
 import { head } from 'lodash-unified'
-import { useEffect, useState } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 import { MaskMessages } from '../../../../../utils'
 import { NextIDProof } from '@masknet/web3-providers'
 import type { Account } from '../type'
+import { initialPersonaInformation } from './PersonaContextInitialData'
 
+function useSSRPersonaInformation() {
+    const [personas, setPersonas] = useState(initialPersonaInformation)
+    const revalidate = useCallback(
+        () => void Services.Identity.queryOwnedPersonaInformation(false).then(setPersonas),
```

Comment:
Race condition here.

Reply:
Can you explain where is the race condition?

Reply:
`revalidate` multiple times.

Reply:
I still don't see the race condition.

```ts
    useEffect(() => void initialPersonaInformation ?? revalidate(), [])
    // ^ this one only fetch data on mount when there is no SSR data
    useEffect(() => MaskMessages.events.ownPersonaChanged.on(revalidate), [])
    // ^ this one is normal event listener.
```